src/complex/DataStructure/DataArray.hpp

```diff
@@ -166,7 +166,7 @@ class DataArray : public IDataArray
 
   /**
    * @brief Returns a deep copy of the DataArray including a deep copy of the
-   * data store.
+   * data store. The object will be owned by the DataStructure.
```

Comment:
It will be owned by DataStructure  as the return statement creates a "new" DataArray(DataStructure).. 

And yes it probably is worth it to review the codes.

Reply:
The `DataArray` constructor doesn't actually add to the `DataStructure` though as far as I can tell. It only stores a reference. It's the static function `DataArray::Create` which actually adds to the `DataStructure`.